src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +53,18 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            for (final Messages.DimensionEntry dimensionEntry : metricData.getDimensionsList()) {
+                final String k = dimensionEntry.getKey();
+                if (!(k.equals(AggClientConnection.CLUSTER_KEY)
```

Comment:
It's a map, so the keys should be unique. Sorting however is required, and I think it's fair to say required in this commit for it to be merged. Bad things will happen otherwise.